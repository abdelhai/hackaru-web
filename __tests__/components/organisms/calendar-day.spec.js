import { Store } from 'vuex-mock-store';
import { shallowMount } from '@vue/test-utils';
import { parse } from 'date-fns';
import CalendarDay from '@/components/organisms/calendar-day';

describe('CalendarDay', () => {
  let wrapper;

  const $store = new Store({
    getters: {
      'activities/getCalendar': () => [
        [
          {
            id: 1,
            description: '',
            startedAt: '2019-01-01T01:30:00',
            stoppedAt: '2019-01-01T02:30:00',
            duration: 3600,
            project: null
          }
        ],
        [
          {
            id: 2,
            description: 'Create a test.',
            startedAt: '2019-01-01T01:00:00',
            stoppedAt: '2019-01-01T02:00:00',
            duration: 3600,
            project: {
              id: 2,
              name: 'Development',
              color: '#ff0'
            }
          }
        ]
      ]
    }
  });

  const factory = () =>
    shallowMount(CalendarDay, {
      mocks: {
        $store,
        $mezr: {
          offset: () => ({
            top: 50
          })
        }
      },
      propsData: {
        day: parse('2019-01-01')
      }
    });

  beforeEach(() => {
    $store.reset();
  });

  describe('when drag ghost-activity', () => {
    beforeEach(() => {
      wrapper = factory();
      wrapper.setData({ ghostHeight: 20 });
      wrapper
        .find('.resizer')
        .vm.$emit('start', { preventDefault: () => {}, pageY: 100 });
    });

    it('show ghost-activity', () => {
      expect(wrapper.find('.ghost-activity').isVisible()).toBe(true);
    });

    it('emit dragging', () => {
      expect(wrapper.emitted('dragging')[0][0]).toEqual({
        el: wrapper.element,
        guideRulerTop: 100 - 50 + 20
      });
    });
  });

  describe('when dragging ghost-activity', () => {
    beforeEach(() => {
      wrapper = factory();
      wrapper.setData({ ghostTop: 50, ghostHeight: 20 });
      wrapper.find('.resizer').vm.$emit('resizing');
    });

    it('emit dragging', () => {
      expect(wrapper.emitted('dragging')[0][0]).toEqual({
        el: wrapper.element,
        guideRulerTop: 50 + 20
      });
    });
  });

  describe('when drop ghost-activity', () => {
    beforeEach(() => {
      wrapper = factory();
      wrapper.setData({ ghostTop: 60, ghostHeight: 20 });
      wrapper.find('.resizer').vm.$emit('end');
    });

    it('emit drop', () => {
      expect(wrapper.emitted('drop')).toBeTruthy();
    });

    it('hide ghost-activity', () => {
      expect(wrapper.find('.ghost-activity').isVisible()).toBe(false);
    });

    it('dispatch activities/addActivity', () => {
      expect($store.dispatch).toHaveBeenCalledWith('activities/addActivity', {
        startedAt: parse('2019-01-01T01:00:00'),
        stoppedAt: parse('2019-01-01T01:20:00')
      });
    });
  });
});
