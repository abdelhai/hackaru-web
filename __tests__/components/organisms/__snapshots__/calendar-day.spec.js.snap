// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay render correctly 1`] = `
<section>
  <div
    class="events"
  >
    <div
      class="row"
    >
      <!---->
    </div>
     
    <div
      class="row"
    >
      <calendar-activity-stub
        day="Tue Jan 01 2019 00:00:00 GMT+0900 (GMT+09:00)"
        description="Create a database."
        duration="3600"
        getoverlapday="() => {}"
        id="1"
        minheight="20"
        project="[object Object]"
        startedat="2019-01-01T01:00:00"
        stoppedat="2019-01-01T02:00:00"
        updateguideline="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
      />
      <calendar-activity-stub
        day="Tue Jan 01 2019 00:00:00 GMT+0900 (GMT+09:00)"
        description=""
        duration="3600"
        getoverlapday="() => {}"
        id="2"
        minheight="20"
        startedat="2019-01-01T01:30:00"
        stoppedat="2019-01-01T02:30:00"
        updateguideline="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
      />
    </div>
    <div
      class="row"
    >
      <calendar-activity-stub
        day="Tue Jan 01 2019 00:00:00 GMT+0900 (GMT+09:00)"
        description="Create a test."
        duration="3600"
        getoverlapday="() => {}"
        id="3"
        minheight="20"
        project="[object Object]"
        startedat="2019-01-01T01:00:00"
        stoppedat="2019-01-01T02:00:00"
        updateguideline="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
      />
    </div>
  </div>
</section>
`;
